Chess/src/pieces/King.java

```diff
@@ -209,4 +246,162 @@ else if (state[x][i].getpiece().getcolor() == this.getcolor())
         }
         return false;
     }
+    
+    public boolean isindangergeneralized(Cell state[][], int x, int y) {
```

Comment:
It may be better to make this `public boolean isindanger(Cell state[], int x, int y)`, and have the first method call it to avoid the repetition:
```java
public boolean isindanger(Cell state[][]) {
    return isindanger(state, x, y);
}